// ngx-socket-io/esm2015/src/socket-io.module.js
import * as ɵngcc0 from '@angular/core';
var SocketIoModule_1;
import * as tslib_1 from "tslib";
import { NgModule, InjectionToken } from '@angular/core';
import { WrappedSocket } from './socket-io.service';
/** Socket factory */
export function SocketFactory(config) {
    return new WrappedSocket(config);
}
export const SOCKET_CONFIG_TOKEN = new InjectionToken('__SOCKET_IO_CONFIG__');
let SocketIoModule = SocketIoModule_1 = class SocketIoModule {
    static forRoot(config) {
        return {
            ngModule: SocketIoModule_1,
            providers: [
                { provide: SOCKET_CONFIG_TOKEN, useValue: config },
                {
                    provide: WrappedSocket,
                    useFactory: SocketFactory,
                    deps: [SOCKET_CONFIG_TOKEN]
                }
            ]
        };
    }
};
SocketIoModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SocketIoModule });
SocketIoModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SocketIoModule_Factory(t) { return new (t || SocketIoModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SocketIoModule, [{
        type: NgModule,
        args: [{}]
    }], null, null); })();
export { SocketIoModule };
